gnuradio-runtime/lib/flat_flowgraph.cc

```diff
@@ -411,24 +411,31 @@ std::string flat_flowgraph::msg_edge_list()
 void flat_flowgraph::dump()
 {
     for (edge_viter_t e = d_edges.begin(); e != d_edges.end(); e++)
-        d_logger->info(" edge: {}", *e);
+        d_logger->info(" edge: {}", (*e).identifier());
 
     for (basic_block_viter_t p = d_blocks.begin(); p != d_blocks.end(); p++) {
-        d_logger->info(" block: {}", *p);
+        d_logger->info(" block: {}", (*p)->identifier());
         block_detail_sptr detail = cast_to_block_sptr(*p)->detail();
-        d_logger->info(" detail @{}:", detail);
+        d_logger->info(" detail @{}:", static_cast<void*>(detail.get()));
 
         int ni = detail->ninputs();
         int no = detail->noutputs();
         for (int i = 0; i < no; i++) {
             buffer_sptr buffer = detail->output(i);
-            d_logger->info("   output {:d}: {}", i, buffer);
+            d_logger->info(
+                "   output {:d}: {}",
+                i,
+                static_cast<std::stringstream&&>(std::stringstream() << buffer).str());
```

Comment:
Traced this through to `operator<<` in `buffer.cc` and it looks sensible. Could you just verify that this previously printed that content and not just a pointer?

Reply:
I was also confused here at first, because the other place with @([this code](https://github.com/gnuradio/gnuradio/pull/6053/files#diff-c21cea5b87f37199c00e6b3de48931e5dd3704d4a98250b7f59e896bf100305bR419)) prints a pointer. Later, after careful analysis and debugging, it was found that the previous behavior here was indeed printing the buffer content.

Reply:
Was it printing buffer content, or the result of `operator<<`?

Reply:
Ah, sorry for the inaccuracy described earlier, its behavior is to print the result of ``operator<<``
E.g.:
```
  sz:    <size>
  nrdrs: <size of reader>
    rd_idx: <read index>
    abs_rd_offset: <read offset>
    rd_idx: <read index>
    abs_rd_offset: <read offset>
    rd_idx: <read index>
    abs_rd_offset: <read offset>
    rd_idx: <read index>
    abs_rd_offset: <read offset>
...
    rd_idx: <read index>
    abs_rd_offset: <read offset>
```